modules/sov-ibc-transfer/src/context.rs

```diff
@@ -345,18 +351,25 @@ impl<'ws, S: Spec> TokenTransferExecutionContext for IbcTransferContext<'ws, S>
                             initial_balance,
                             minter_address,
                             authorized_minters,
-                            &self.sdk_context,
+                            &context,
                             &mut self.working_set.borrow_mut(),
                         )
                         .map_err(|err| TokenTransferError::Other(err.to_string()))?;
 
-                    // Store the new address in `minted_tokens`
-                    self.ibc_transfer.minted_tokens.set(
+                    // Stores mapping from "denom to token ID" of the IBC-minted token
+                    self.ibc_transfer.minted_token_name_to_id.set(
                         &denom,
                         &new_token_id,
                         *self.working_set.borrow_mut(),
                     );
 
+                    // Stores mapping from "token ID to denom" of the IBC-minted token
+                    self.ibc_transfer.minted_token_id_to_name.set(
+                        &new_token_id,
+                        &denom,
+                        *self.working_set.borrow_mut(),
+                    );
+
```

Comment:
we may want to put this in a method - as these two calls should be atomic.

Reply:
Good suggestion!

Reply:
[b927153](https://github.com/informalsystems/sovereign-ibc/pull/133/commits/b9271531b1b259510603b1c0048725a82dceea78)